From ff396d09cbd60aa4945d056c6f9880bddd35cc8d Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Fri, 05 Oct 2018 17:47:51 +0300
Subject: [PATCH] Makefile: release: add hl-check-branch target

The new target is supposed to check that all the repositories listed
in RELEASE_REPOS are aligned with the release branch.
It checks if we are on the upstream release branch.
Then it walks through the repositories in RELEASE_REPOS and checks
that each repository's HEAD is pointed by the upstream release branch.
If there is an unaligned repository, the release target won't be
executed.

Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/Makefile b/Makefile
index b583626..8b5921a 100644
--- a/Makefile
+++ b/Makefile
@@ -81,12 +81,33 @@
 RELEASE_JOB_TOKEN := "token=goya-release"
 RELEASE_JOB_PARAMS := $(RELEASE_JOB_TOKEN)&$(RELEASE_JOB_CAUSE)&$(RELEASE_JOB_BRANCH)&$(RELEASE_VER_COMPAT)
 
+define NO_RELEASE_BRANCH_ERROR =
+Assuming the upstream release branch looks like 'origin/vx.y.z'.
+Your repository is not alinged with upstream release branch
+endef
+
+.PHONY: hl-check-branch
+hl-check-branch:
+ifeq ($(RELEASE_BRANCH),)
+	$(error $(NO_RELEASE_BRANCH_ERROR))
+else
+	@for dir in $(RELEASE_REPOS); do \
+		pushd ../$$dir > /dev/null; \
+		if [ $$(git branch -r --points-at HEAD | grep -cE "origin/$(RELEASE_BRANCH)") -lt 1 ]; then \
+			echo "$$dir - is not aligned with origin/$(RELEASE_BRANCH)"; \
+			ALL_REPOS_ALIGNED=false; \
+		fi; \
+		popd > /dev/null; \
+	done; \
+	$$ALL_REPOS_ALIGNED
+endif
+
 .PHONY: print-release-repos
 print-release-repos:
 	@echo $(RELEASE_REPOS)
 
 .PHONY: gaudi-release goya-release
-gaudi-release goya-release: hl-check-release
+gaudi-release goya-release: hl-check-release hl-check-branch
 	$(eval EMBD_FULL_VER=hl-$(patsubst %-release,%,$@)-$(EMBD_RELEASE_VER))
 	$(eval RELEASE_JOB_PROD_NAME="product_name=$(patsubst %-release,%,$@)")
 	$(eval RELEASE_JOB_TAG_PTR="embedded_release_tag=$(EMBD_FULL_VER)")
